share/p2p/shrexnd/client.go

```diff
@@ -55,7 +55,7 @@ func (c *Client) RequestND(
 	peer peer.ID,
 ) (share.NamespacedShares, error) {
 	shares, err := c.doRequest(ctx, root, nID, peer)
-	if err == nil {
+	if err == nil || errors.Is(err, share.ErrNamespaceNotFound) {
```

Comment:
When will it hit this instead of line 122?

Reply:
If 122 is not hit, it will hit no error case. Both needs to return data + error